### ISSUE TITLE
Do something about loose LuaUI files in default engine installation

### ISSUE THREAD

**Issue by sprunk** [beyond-all-reason/spring#1483]
The engine installation comes with a bunch of loose LuaUI files (these https://github.com/beyond-all-reason/spring/tree/BAR105/cont/LuaUI ).

This presents two problems:
* games rely on those files, which is fragile because they are undocumented and just happen to work. But players can just delete or overwrite them etc., it's just a ticking distribution bomb. They should be in some sort of basecontent-level archive for checksum etc.
* many of these files are of very low quality. So just dumping them in an archive would waste a good opportunity to improve them.

The first task here is probably to make sure that existing games won't break if we touch these files.

**Comment by springraaar** [comment]
A few days ago i noticed my game's copies of these files were mostly unchanged and deleted them. I thought they were part of basecontent but they're accessible on VFS.RAW instead. I had trouble overriding selector.lua because of this.

Moving them from raw to basecontent would probably let games loading files using RAW_FIRST to override them while giving users the possibility to override them. 

> Apparently the loose files are not enough for an empty mod to recognize a widget placed in ./luaui/widgets out of the box, which is bad.

luaui vs LuaUI . VFS.RAW is case-sensitive on linux but not windows, the others seem case-insensitive.

**Comment by badosu** [comment]
At some point we need to discuss the epic "basecontent is not too usable for new games at all" (do we have an issue for that?), the following would be relevant in that discussion:

1. LuaRules basecontent in the sense of gadget manager (and included base gadgets) is exposed as `LuaGadgets` folder
2. It would make sense for LuaUI basecontent in the sense of widget manager (and included base widgets) to be exposed as `LuaWidgets` folder.
3. Is 1. and 2. even reasonable at all, it's not like having an empty file requiring from `LuaGadgets` is too usable for new games, it's so cryptic, would it make sense to have some other approach to be blessed by Recoil? (e.g. options either/or: vendor all this shit, make it your own - your shit gets overwritten by default vendor it to not be overwritten - allow basecontent to not be included by default in ALL games)
4. I guess LuaUI never made it because effort was put into making `LuaHandler` the true manager for all envs, no more duplication of "wupget/addon" handling logic, but that effort never materialized into more than just current luamenu and luaintro envs.

**Comment by SlashScreen** [comment]
for the record basecontent does not sound like "default implementation"

**Comment by badosu** [comment]
> for the record basecontent does not sound like "default implementation"

Unclear what's this statement about, what does "default implementation" mean?

**Comment by SlashScreen** [comment]
Isn't the purpose of basecontent to be a default/built-in implementation of the basic gadgets, widgets, handlers, etc?

**Comment by badosu** [comment]
Not sure there's an explicit purpose other than provide a base dependency for games (as it does right now).

Often canonical implementations (in the sense of as stripped of ad-hoc behavior as possible) would make sense to exist in basecontent (if that would be "default" I guess), but lack of maintenance made many aspects of basecontent outdated (and up to test possibly many non-functional pieces of code).

**Comment by SlashScreen** [comment]
whatever it's called, luaui should be moved into base content.

**Comment by badosu** [comment]
Yes, that's the whole point of this issue

**Comment by SlashScreen** [comment]
Right, yes, sorry. I meant to ask why we can't do that. I don't know why I expected you to read my mind.

**Comment by SlashScreen** [comment]
What about having a clause in `main.lua` that has you disable basecontent?
